assets/dev/js/editor/components/template-library/views/parts/templates.js

```diff
@@ -339,6 +341,22 @@ const TemplateLibraryCollectionView = Marionette.CompositeView.extend( {
 		this.removeScrollListener = () => scrollableContainer.off( 'scroll', listener );
 	},
 
+	onCreateNewFolderClick() {
+		const activeSource = elementor.templates.getFilter( 'source' );
+
+		if ( 'cloud' !== activeSource ) {
+			return;
+		}
+
+		elementor.templates.createFolder( {
+			source: activeSource,
+		},
+		{
+			onSuccess: () => {
+				elementor.templates.showTemplates();
```

Comment:
```suggestion
				$e.routes.refreshContainer( 'library' );
```
Following on my [previous comment](https://github.com/elementor/elementor/pull/30326/files#r1970791907), it seems like this actually works better.

Reply:
We'll need to apply this fix on the delete action as well in Local and Cloud.

Reply:
Done but it feels like a workaround and not like solving the bug.